src/qibo/tensorflow/distutils.py

```diff
@@ -404,7 +404,9 @@ def default(cls, circuit: "DistributedCircuit"):
       """Creates the |000...0> state for default initialization."""
       state = cls(circuit)
       with tf.device(state.device):
-          state.pieces[0] = op.initial_state(state.pieces[0].shape, state.pieces[0].dtype, get_threads())
+          state.pieces[0] = op.initial_state(nqubits=state.nlocal,
+                                             dtype=DTYPES.get('DTYPECPX'),
+                                             is_matrix=False, omp_num_threads=get_threads())
```

Comment:
We can leave all the `.assign` calls as they are and change this to:
```suggestion
          piece = op.initial_state(nqubits=state.nlocal,
                                   dtype=DTYPES.get('DTYPECPX'),
                                   is_matrix=False, omp_num_threads=get_threads())
         state.pieces[0] = tf.Variable(piece, dtype=piece.dtype)
```
This way all tests pass and performance is the same as master.

Reply:
Thanks for spotting this issue. 

However, I had to change all `assign` with `=` because, at least on CPU, the code crashes with:
```     
>               state.pieces[i].assign(piece)
E               AttributeError: 'tensorflow.python.framework.ops.EagerTensor' object has no attribute 'assign'
```

Does this happens to you too?

Reply:
If I do the above change `.assign` works both on CPU and GPU. I think changing the initialization as I wrote above using `tf.Variable(op.initial_state(...))` will solve this crash because all `.assign` calls will happen on variables not tensors.

Reply:
Thanks, let me try to revert all assigns after applying your initialization.